### ISSUE TITLE
Documentation XMonad.Hooks.DynamicBars

### ISSUE THREAD

**Issue by vherrmann** [xmonad/xmonad-contrib#593]
### Problem Description

Imho the documentation for XMonad.Hooks.DynamicBars could use some examples. I will try to fix this myself, but I don't have much time atm.

### Checklist

  - [ ] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - I tested my configuration
    - [ ] With `xmonad` version XXX (commit XXX if using git)
    - [ ] With `xmonad-contrib` version XXX (commit XXX if using git)

**Comment by liskin** [comment]
Actually, we have https://xmonad.github.io/xmonad-docs/xmonad-contrib-0.16.999/XMonad-Hooks-StatusBar.html#g:4 now so we might as well deprecate DynamicBars in the future instead of fixing the docs. Can you give X.H.StatusBar a try and check if it satisfies all your needs?

**Comment by vherrmann** [comment]
Indeed, this satifies (as far as i'm aware) my needs. Apart from that, a deprecated warning would be nice for XMonad.Hooks.DynamicBars.

**Comment by slotThe** [comment]
> EDIT: Since XMonad.Hooks.StatusBar doesn't seem to be (at least) in nixpkgs, it's a little early to depreciate X.H.DyamicBars yet

The rule is to keep deprecations around for 2--3 releases, so people have more than enough time to switch from X.H.DynamicBars to the new variants

**Comment by TheMC47** [comment]
So should we close the issue? I do agree that it was very difficult to figure out how `X.H.DynamicBars` can be used, but it's not worth the effort to update the docs

@vherrmann did you figure out how to use `X.H.StatusBar`? If you're compiling from git (which is a very sensible thing to do) then you can tell us if you get stuck on anything or if anything is unclear. If you're not (and don't want to) I could probably write an example using `X.H.DynamicBars`

**Comment by liskin** [comment]
A PR that marks DynamicBars deprecated in favor of X.H.StatusBar and keeps them around for a couple releases would be the best way to close this issue.

**Comment by vherrmann** [comment]
@TheMC47 
I did figure out how to use X.H.StatusBar, though i had big trouble using the xmonad flake in my nixos system configuration, now it does work, but i have no idea why :S; Another pretty weird issue was that `dynamicBSs` didn't spawn the bars until I myself spawned them manually in the terminal but then it just worked…. Apart from all that, the config works like a charm, and it also simplifies my previous setup.

**Comment by TheMC47** [comment]
> Another pretty weird issue was that `dynamicBSs` didn't spawn the bars until I myself spawned them manually in the terminal but then it just worked

can you elaborate on that?

**Comment by prikhi** [comment]
> Can you give X.H.StatusBar a try and check if it satisfies all your needs?

I have a weird usecase & am not sure if I can replicate it w/ the StatusBar module. Only done a cursory look over the module though.

Essentially, I use `forkProcess` to launch `Xmobar.App.Main.xmobar` & communicate w/ it via it's `stdin` `Handle`(instead of a pipe's `Handle`). So there's no "executable" to pass to the various `statusBarXYZ` functions:

https://github.com/prikhi/xmonad-config/blob/master/src/XmobarStub.hs#L13-L27
https://github.com/prikhi/xmonad-config/blob/master/src/StatusBar.hs#L22-L61
https://github.com/prikhi/xmonad-config/blob/master/src/Config.hs#L185-L250

Maybe possible my making my own custom `StatusBarConfig` with `dynamicSBs`?

**Comment by TheMC47** [comment]
> Maybe possible my making my own custom `StatusBarConfig` with `dynamicSBs`?

I think this is very well possible. The `statusBar*` are just helper functions to create suitable `StatusBar`s. The type `StatusBar` is deliberately general enough to accommodate for other exotic use cases: you just need to tell how to spawn the bar, how to kill it, and how to send data to it. And since these are `X ()`, you can essentially go wild and send a rocket to the moon. If you respect the contract (i.e. `sbCleanupHook` kills the bar, `sbStartupHook` starts the bar, and `sbLogHook` sends data to the bar) then you can make your bars dynamic with `dynamicSBs`

**Comment by TheMC47** [comment]
#613 deprecates the module, and I think anyone using it would figure out how to use the simpler `StatusBar` (at least for basic usages). Closing this for now.
